drv/lpc55-update-server/src/main.rs

```diff
@@ -17,12 +17,21 @@ use drv_lpc55_update_api::{
 };
 use drv_update_api::UpdateError;
 use idol_runtime::{ClientError, Leased, LenLimit, RequestError, R, W};
+use ringbuf::*;
 use stage0_handoff::{
-    HandoffData, HandoffDataLoadError, ImageVersion, RotBootState,
+    HandoffData, HandoffDataLoadError, ImageError, ImageVersion, RotBootState,
 };
 use userlib::*;
 use zerocopy::{AsBytes, FromBytes};
 
+#[derive(Copy, Clone, PartialEq)]
+enum Trace {
+    None,
+    Slot(u32, Result<ImageVersion, ImageError>),
+    NoBootstate(HandoffDataLoadError),
+}
+ringbuf!(Trace, 64, Trace::None);
+
```

Comment:
How much space do we save without the ringbuf? We're not close to any limits but it's not clear how valuable this may be for a production system